pyanaconda/payload/dnfpayload.py

```diff
@@ -1109,16 +1138,22 @@ def updateBaseRepo(self, fallback=True, checkmount=True):
 
             self._add_repo(ksrepo)
 
-        ksnames = [r.name for r in self.data.repo.dataList()]
-        ksnames.append(constants.BASE_REPO_NAME)
         with self._repos_lock:
+
+            # disable unnecessary repos
             for repo in self._base.repos.iter_enabled():
                 id_ = repo.id
                 if 'source' in id_ or 'debuginfo' in id_:
                     self.disableRepo(id_)
                 elif constants.isFinal and 'rawhide' in id_:
                     self.disableRepo(id_)
 
+            # fetch md for enabled repos
+            enabled_repos = self.enabledRepos
+            for repo in self.addOns:
+                if repo.name in enabled_repos:
+                    self._fetch_md(repo)
```

Comment:
Couldn't this fetch the metadata second time for some repos? I guess it is not such a problem but still.

Reply:
That is a bit of a puzzle....